### PR TITLE
Version 1 of optimizing the CI

### PR DESCRIPTION
**Pull request by NadeemYaseen**
Create only one release build for the batch tests. Divide the batch test into groups to run them in parallel using the matrix strategy
Create only one debug build for all the debug tests and run it only on OSFPA to save time.
Combine the common steps like dependencies installation, setting compiler, device type, etc for all types of build into one use action
Divide the dependencies into two sections that are minimum and full. This allows us to choose only the minimum required ones to run the compiled code.
 Separate Makefile for all the test-related commands. Introduce some new Makefile targets to remove the commands from the CI YAML file to the Makefile
Each type of test group needs a specific type of build. The builds are separate jobs so that the test group can start running as soon as possible.

### CODE REVIEW

**Review by alaindargelas — APPROVED** [review]

**Comment by alaindargelas** [comment]
Last PR before this one ran in [2h 25m 41s](https://github.com/os-fpga/Raptor/actions/runs/11099360599/usage).
Something wrong here, seems something is stuck

**Comment by ravikiranchollangi** [comment]
> @NadeemYaseen, Last PR before this one ran in [2h 25m 41s](https://github.com/os-fpga/Raptor/actions/runs/11099360599/usage). Something wrong here, seems something is stuck

We will sync up later today, it appears to be a GH runner availability issue from logs.

**Comment by NadeemYaseen** [comment]
> > @NadeemYaseen, Last PR before this one ran in [2h 25m 41s](https://github.com/os-fpga/Raptor/actions/runs/11099360599/usage). Something wrong here seems something is stuck
> 
> We will sync up later today, it appears to be a GH runner availability issue from the logs.

You are correct. We can see the actual run time of any run by going to usage. For example, in the below screen, the time may appear like 24 minutes as shown in below screenshot, but if you click on [usage](https://github.com/os-fpga/Raptor/actions/runs/11136028311/usage), it shows actual time that is 7 seconds which is correct as we are only pinging a API in this YAML file and 24 minutes seems unreasonable to make a ping.
image

File: examples/and2_gemini/and2_gemini.ospr [hunk 1]
```diff
@@ -49,8 +49,8 @@
         <Run Name="imple_1" Type="Implementation" SrcSet="sources_1" ConstrsSet="constrs_1" State="current" SynthRun="synth_1"/>
         <Run Name="synth_1" Type="Synthesis" SrcSet="sources_1" ConstrsSet="constrs_1" State="current" SynthRun="">
             <Option Name="Compilation Flow" Val="Classic Flow"/>
-            <Option Name="Device" Val="1VG28"/>
-            <Option Name="Family" Val="Virgo"/>
+            <Option Name="Device" Val="MPW1"/>
+            <Option Name="Family" Val="Internal-Gemini"/>
```
Comment (ravikiranchollangi) [inline examples/and2_gemini/and2_gemini.ospr:53]: Revert this.

Comment (ravikiranchollangi) [reply examples/and2_gemini/and2_gemini.ospr:53]: It also takes 6HRs to complete on GH CI.

File: tests/TestGui/gui_run_and2_mpw1_project.tcl [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#Copyright 2021 The Foedag team
```
Comment (ravikiranchollangi) [inline tests/TestGui/gui_run_and2_mpw1_project.tcl:1]: Revert this too.

**Review by ravikiranchollangi — COMMENTED** [review]
Reverts needed.

**Pull request merged**